### ISSUE TITLE
[15.0] groups attribute is ignored in list view with the states attribute

### ISSUE THREAD

**Issue by adrienpeiffer** [odoo/odoo#99770]
Impacted versions:

15.0

Groups attributes are evaluated here :+1: 

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by yelizariev** [comment]
It looks as a bug indeed.

The view from the video:

https://github.com/odoo/odoo/blob/dffa40f744398511a9ad1eb1c3bf8b8d76e2b6d8/addons/hr_holidays/views/hr_leave_views.xml#L531-L538

**Comment by yelizariev** [comment]
This problem is fixed in master (v16), where groups are applied at the last moment: https://github.com/odoo/odoo/commit/1744870d8de56d930c5dfae892a70b25098070b7
 
Potential solution for v15 could be changing order of calls `_apply_groups` and `transfer_node_to_modifiers`.

https://github.com/odoo/odoo/blob/5e6f4d395dc170778bbfecac3b7ba9bef55561f5/odoo/addons/base/models/ir_ui_view.py#L1066-L1067

However, it will change visibility of existing views, which is not something that users may expect. Imagine a company with a  "Time off officier" user. They do Validation without knowing that they need Administrator access level for that. Then one day they open Time Off menu and find no Validate/Refuse buttons.

For this reason we can't fix it in stable. If you need to use states with groups on your custom modules, you should try some workarounds (e.g. add such buttons via extension view with field `groups_id`)

PS I'd keep this issue open to track this issue, post more info on workarounds, etc.

**Comment by adrienpeiffer** [comment]
Yes, It'll change the visibility of existing views because it's the bug. For the moment, the user which hasn't the right to validate the leave can do it because of this bug.

**Comment by adrienpeiffer** [comment]
Replace states attribute by attrs should make the job. Please fix it.

**Comment by Malek907** [comment]
Hello , any news about this bug ?

**Comment by Malek907** [comment]
@Yenthe666 Can you check this bug please
